mediapipe/graphs/instantmotiontracking/subgraphs/calculators/tracked_anchor_manager_calculator.cc

```diff
@@ -0,0 +1,154 @@
+private:
+  std::vector<Anchor> previous_anchor_data_;
+  const float box_h_w_ = 0.5; // Used to establish tracking box dimensions
+
+public:
+  static ::mediapipe::Status GetContract(CalculatorContract* cc) {
+    RET_CHECK(!cc->Inputs().GetTags().empty());
+    RET_CHECK(!cc->Outputs().GetTags().empty());
+
+    if(cc->Inputs().HasTag(kAnchorsTag)) {
+      cc->Inputs().Tag(kAnchorsTag).Set<std::vector<Anchor>>();
+    }
+    if(cc->Inputs().HasTag(kBoxesInputTag)) {
+      cc->Inputs().Tag(kBoxesInputTag).Set<TimedBoxProtoList>();
+    }
+    if(cc->Outputs().HasTag(kBoxesOutputTag)) {
+      cc->Outputs().Tag(kBoxesOutputTag).Set<TimedBoxProtoList>();
+    }
+    if(cc->Outputs().HasTag(kAnchorsTag)) {
+      cc->Outputs().Tag(kAnchorsTag).Set<std::vector<Anchor>>();
+    }
+    if(cc->Outputs().HasTag(kCancelTag)) {
+      cc->Outputs().Tag(kCancelTag).Set<int>();
+    }
+
+    return ::mediapipe::OkStatus();
+  }
+
+  ::mediapipe::Status Open(CalculatorContext* cc) override {
+    cc->SetOffset(TimestampDiff(0));
+    return ::mediapipe::OkStatus();
+  }
+
+  ::mediapipe::Status Process(CalculatorContext* cc) override;
+};
+REGISTER_CALCULATOR(TrackedAnchorManagerCalculator);
+
+::mediapipe::Status TrackedAnchorManagerCalculator::Process(CalculatorContext* cc) {
+  std::vector<Anchor> current_anchor_data = cc->Inputs().Tag(kAnchorsTag).Get<std::vector<Anchor>>();
+
+  auto pos_boxes = absl::make_unique<TimedBoxProtoList>();
+
+  std::vector<Anchor> tracked_scaled_anchor_data;
+
+  // Update from tracking system or add anchor to tracking system
+  for(const Anchor previous_anchor : previous_anchor_data_) {
+    for(Anchor current_anchor : current_anchor_data) {
+      if(previous_anchor.sticker_id == current_anchor.sticker_id) {
+        // Check if anchor was repositioned by user, and reset tracking box
+        if(previous_anchor.x != current_anchor.x || previous_anchor.y != current_anchor.y) {
+          TimedBoxProto* box = pos_boxes->add_box();
+          box->set_left(current_anchor.x - box_h_w_/2);
+          box->set_right(current_anchor.x + box_h_w_/2);
+          box->set_top(current_anchor.y - box_h_w_/2);
+          box->set_bottom(current_anchor.y + box_h_w_/2);
+          box->set_id(current_anchor.sticker_id);
+          box->set_time_msec(cc->InputTimestamp().Microseconds() / 1000);
+          current_anchor.z = 1.0; // Default value for normalized z (scale factor)
+        }
+        // If anchor was not repositioned, update the location from the tracking system
+        else if (cc->Inputs().HasTag(kBoxesInputTag) && !cc->Inputs().Tag(kBoxesInputTag).IsEmpty()) {
+          for(const auto& box : cc->Inputs().Tag(kBoxesInputTag).Get<TimedBoxProtoList>().box())
+          {
+            if(box.id() == current_anchor.sticker_id) {
+              current_anchor.x = (box.left() + box.right())/2;
+              current_anchor.y = (box.top() + box.bottom())/2;
+              current_anchor.z = box_h_w_/(box.right() - box.left());
```

Comment:
Let's add a quick comment for how we get this z-value (specifically let's mention that this assumes the z-value starts off normalized to 1, and scales inversely with box-width).  This method is potentially a bit problematic if the box doesn't scale uniformly on y-axis and x-axis, but that's something we can look into later if necessary.